src/transformers/modeling_utils.py

```diff
@@ -327,6 +327,63 @@ def get_checkpoint_shard_files(
     return cached_filenames, sharded_metadata
 
 
+def load_sharded_checkpoint(model, folder, strict=True):
+    """
+    This is the same as
+    [`torch.nn.Module.load_state_dict`](https://pytorch.org/docs/stable/generated/torch.nn.Module.html?highlight=load_state_dict#torch.nn.Module.load_state_dict)
+    but for a sharded checkpoint.
```

Comment:
I think it would be nice to have a documentation for sharded checkpoints so that users understand what they are. For example here having the `sharded checkpoint` redirect to a small blurb mentioning things like the following:
- What is it?
    - Weights files that are split in multiple checkpoint
    - Index showing how weights are linked
- Why is it important?
    - Better to work with smaller files for memory
    - Simpler to push to the hub 
- How to work with it?
    - Showing how to use `from_pretrained` and `save_pretrained` for sharding
    - push to hub
    - now trainer

Let me know if that's something that already exists, and if not I'm happy to help contribute it (or to contribute it altogether).